src/prototype_room_utils.js

```diff
@@ -1,5 +1,44 @@
 'use strict';
 
+Room.prototype.sortMyRoomsByLinearDistance = function(target) {
+  let sortByLinearDistance = function(object) {
+    return Game.map.getRoomLinearDistance(target, object);
+  };
+
+  return _.sortBy(Memory.myRooms, sortByLinearDistance);
+};
+
+Room.prototype.closestSpawn = function(target) {
+  let pathLength = {};
+  let roomsMy = this.sortMyRoomsByLinearDistance(target);
+
+  for (let room of roomsMy) {
+    let route = Game.map.findRoute(room, target);
+    let routeLength = global.utils.returnLength(route);
+
+    if (route && routeLength) {
+      //TODO @TooAngel please review: save found route from target to myRoom Spawn by shortest route!
+      //Memory.rooms[room].routing = Memory.rooms[room].routing || {};
+      //Memory.rooms[room].routing[room + '-' + target] = Memory.rooms[room].routing[room + '-' + target] || {
+      //    path: room + '-' + route,
+      //    created: Game.time,
+      //    fixed: false,
+      //    name: room + '-' + target,
+      //    category: 'moveToByClosestSpawn'
+      //  };
```

Comment:
Hm, not yet sure.

So it is mainly a CPU optimization. If, it should be before `findRoute` to save CPU :-)
I don't think it will we called too often.
To properly save CPU we should store the complete route, so that the creeps could get the route from memory.
I would also suggest to change it from `Memory.rooms[room].routing` to something else to not collide with the existing, which is for in-room paths. (Better change `.routing` to `.pathing` or similar, but this would be a really big change.)